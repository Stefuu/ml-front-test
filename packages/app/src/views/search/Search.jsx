import React from 'react'
import SearchBar from '../../common/SearchBar'
import { withSearchBar } from '../../common/SearchBarContext'

class Search extends React.Component {
  render () {
    return (
      <SearchBar
        onSubmit={this.props.context.onSubmit}
        onChange={this.props.context.onChange}
        searchTerm={this.props.context.searchTerm}
      />
    )
  }
}

export default withSearchBar(Search)
