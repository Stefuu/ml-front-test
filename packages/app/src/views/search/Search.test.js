import React from 'react'
import theme from '../../components/themes/default'
import { ThemeProvider } from 'styled-components'
import { render, fireEvent } from 'react-testing-library'
import { BrowserRouter } from 'react-router-dom'
import GlobalStyle from '../../components/modules/GlobalStyle'
import SearchBarProovider from '../../components/contexts/SearchBarContext'
import Search from './Search'

const Component = () => (
  <BrowserRouter>
    <ThemeProvider theme={theme}>
      <SearchBarProovider>
        <div style={{ width: '100vw' }}>
          <GlobalStyle />
          <Search />
        </div>
      </SearchBarProovider>
    </ThemeProvider>
  </BrowserRouter>
)

describe('Search', () => {
  it('Should render [snapshot]', async () => {
    const { container } = render(<Component />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('Should be able to type into the input and submit', async () => {
    const { getByTestId } = render(<Component />)
    const input = getByTestId('input-search')
    const searchButton = getByTestId('search-button')
    fireEvent.change(input, { target: { value: 'notebook' } })
    fireEvent.click(searchButton)
    fireEvent.change(input, { target: { value: 'TEST VALUE' } })
    expect(input.value).toBe('TEST VALUE')
    fireEvent.click(searchButton)
    fireEvent.change(input, { target: { value: 'TEST VALUE 2' } })
    expect(input.value).toBe('TEST VALUE 2')
  })
})
