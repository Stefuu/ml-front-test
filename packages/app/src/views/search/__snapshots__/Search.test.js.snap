// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Should render [snapshot] 1`] = `
<div
  style="width: 100vw;"
>
  <div
    class="sc-bdVaJa sc-ifAKCX hUNZUI"
  >
    <div
      class="sc-bdVaJa fEwoCY"
    >
      <div
        class="sc-bdVaJa cyMNyE"
      >
        <a
          class="sc-bdVaJa cpotsO"
          href="/"
          mr="1.6875rem"
        >
          <div
            alt="Logo"
            class="sc-bxivhb jLbKpf"
            height="2.25rem"
            width="3.3125rem"
          />
        </a>
        <form
          class="sc-bdVaJa gGeGEh"
          display="flex"
          width="100%"
        >
          <input
            class="sc-bwzfXH VvuCb"
            data-testid="input-search"
            placeholder="Nunca dejes de buscar"
            type="text"
            value=""
          />
          <button
            class="sc-htpNat kAygXV"
            data-testid="search-button"
          >
            <div
              alt="Logo"
              class="sc-bxivhb hGGnaW"
              height="1.125rem"
              width="1.25rem"
            />
          </button>
        </form>
      </div>
    </div>
  </div>
  <div
    class="sc-bdVaJa sc-EHOje fkUvOS"
  />
</div>
`;
